instrumentation/google.golang.org/grpc/otelgrpc/grpctrace.go

```diff
@@ -135,29 +135,45 @@ func (s *metadataSupplier) Keys() []string {
 // Inject injects correlation context and span context into the gRPC
 // metadata object. This function is meant to be used on outgoing
 // requests.
+// Deprecated: Unnecessary public func.
 func Inject(ctx context.Context, md *metadata.MD, opts ...Option) {
 	c := newConfig(opts)
-	inject(ctx, md, c.Propagators)
+	c.Propagators.Inject(ctx, &metadataSupplier{
+		metadata: md,
+	})
 }
 
-func inject(ctx context.Context, md *metadata.MD, propagators propagation.TextMapPropagator) {
+func inject(ctx context.Context, propagators propagation.TextMapPropagator) context.Context {
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		md = metadata.MD{}
+	}
 	propagators.Inject(ctx, &metadataSupplier{
-		metadata: md,
+		metadata: &md,
 	})
+	return metadata.NewOutgoingContext(ctx, md)
 }
 
 // Extract returns the correlation context and span context that
 // another service encoded in the gRPC metadata object with Inject.
 // This function is meant to be used on incoming requests.
+// Deprecated: Unnecessary public func.
 func Extract(ctx context.Context, md *metadata.MD, opts ...Option) (baggage.Baggage, trace.SpanContext) {
```

Comment:
Decided to mark this funcs as deprecated, I can also remove them if you are ok with that, they should not be used outside but I am not sure about your policy.

For Extract was very interesting that you extract Baggage to add it back.